utils/fileutil/filetutil.go

```diff
@@ -139,38 +139,42 @@ func CompressDir(dir string) error {
 	if err != nil {
 		return err
 	}
-	b := new(bytes.Buffer)
-	zw := zip.NewWriter(b)
+
+	outFileName := fmt.Sprintf("%s.zip", dir)
+	outFilePath := filepath.Join(dir, outFileName)
+
+	outFile, err := os.Create(outFilePath)
+	if err != nil {
+		return err
+	}
+	defer outFile.Close()
+
+	zw := zip.NewWriter(outFile)
+	defer zw.Close()
+
 	for _, f := range files {
 		fw, err := zw.Create(f.Name())
 		if err != nil {
 			return err
 		}
-		name := path.Join(dir, f.Name())
-		content, err := os.ReadFile(name)
+
+		contentPath := filepath.Join(dir, f.Name())
+		contentFile, err := os.Open(contentPath)
 		if err != nil {
 			return err
 		}
-		_, err = fw.Write(content)
+		defer contentFile.Close()
```

Comment:
Using defer statements in a for loop is not a good idea.

Ref: [Effective Go](https://go.dev/doc/effective_go#defer)